test/e2e-test/regression-test/util/interceptors.js

```diff
@@ -0,0 +1,123 @@
+// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+const _ = require('lodash');
+
+class Interceptor {
+    constructor (next) {
+        this.__next = next;
+    }
+
+    handle (data) {
+        if (!!this.__next && this.__next instanceof Interceptor) {
+            return this.__next.handle(data);
+        }
+        return data;
+    }
+}
+
+class DoNothingInterceptor extends Interceptor {
+    handle (data) {
+        return super.handle(data);
+    }
+}
+
+class ExtraDynamicFieldInterceptor extends Interceptor {
+    constructor (next) {
+        super(next);
+        this.__placeholder = 'removed';
+        this.__removeUUIDResourceTypes = [
+            'DocumentReference', 'Composition', 'Immunization', 'MedicationStatement',
+            'Condition', 'Observation'
```

Comment:
Why do we remove UUID for these types?

Reply:
These IDs are all platform-related and will cause the error of NodeJS CI, so we will not compare their IDs while regression testing.

Reply:
Just curious are these IDs platform-related?

Reply:
In the `cda/Section/Immunization.hbs` template, there is a UUID generation statement `ID=(generateUUID (toJsonString immEntry))`. The cda message `immEntry` may contain origin xml data, which may contain platform related line ending characters.